# HG changeset patch
# User sbr <sebastien@brissaud.name>
# Date 1374913741 -7200
#      Sat Jul 27 10:29:01 2013 +0200
# Node ID 81b6cbe1b3ca1982b3ce4f0a437772af1fe6da70
# Parent  1ae4021b468b0cf46c74ad867c578e4a3d8c34da
-Fix [FS#5666] (r25096): TownPopulationSorter function should return the result of an ascending comparison.

diff --git a/src/town_gui.cpp b/src/town_gui.cpp
--- a/src/town_gui.cpp
+++ b/src/town_gui.cpp
@@ -686,13 +686,13 @@ private:
 		return strnatcmp(buf, buf_cache); // Sort by name (natural sorting).
 	}
 
-	/** Sort by population (default descending, as big towns are of the most interest). */
+	/** Sort by population. */
 	static int CDECL TownPopulationSorter(const Town * const *a, const Town * const *b)
 	{
 		uint32 a_population = (*a)->cache.population;
 		uint32 b_population = (*b)->cache.population;
 		if (a_population == b_population) return TownDirectoryWindow::TownNameSorter(a, b);
-		return (a_population > b_population) ? -1 : 1;
+		return (a_population < b_population) ? -1 : 1;
 	}
 
 	/** Sort by town rating */
